freqtrade/freqai/freqai_interface.py

```diff
@@ -147,12 +152,20 @@ def start(self, dataframe: DataFrame, metadata: dict, strategy: IStrategy) -> Da
             dataframe = self.dk.use_strategy_to_populate_indicators(
                 strategy, prediction_dataframe=dataframe, pair=metadata["pair"]
             )
-            dk = self.start_backtesting(dataframe, metadata, self.dk)
+            if not self.save_live_data_backtest:
+                dk = self.start_backtesting(dataframe, metadata, self.dk)
+                dataframe = dk.remove_features_from_df(dk.return_dataframe)
+            else:
+                dk = self.start_backtesting_from_live_saved_files(
+                    dataframe, metadata, self.dk)
+                dataframe = dk.return_dataframe
 
-        dataframe = dk.remove_features_from_df(dk.return_dataframe)
         self.clean_up()
         if self.live:
             self.inference_timer('stop', metadata["pair"])
+            if self.save_live_data_backtest:
+                dk.save_backtesting_live_dataframe(dataframe, metadata["pair"])
```

Comment:
It seems to me you are rebuilding `historical_predicions.pkl`. Is it possible to use `historic_predictions.pkl` directly instead of building/saving more dataframes? Or perhaps `historical_predictions.pkl` is missing a key piece of information? If so, can we simply add that information to `historical_predictions.pkl`? This way we may be able to consolidate machinery. 

Let me know your thoughs.

Reply:
I think it's possible to use `historical_predicions.pkl`.
The main reasons I decided to save the DF again are:

- I can easily get dry/live start and end date. My code is just recording the current candle and `historical_predictions.pkl` brings all historical candles from the dataframe. We can count candle numbers, but I don't know if that's the best way to handle it.
- We have separate files per pair and I think it's better to deal with small files in dry/live to attach current candles. The backtest loop is per pair as well, which can bring more performance.
- The new code is relatively simple and isolated, which can bring more flexibility in the future, in case we have different requirements for the 2 features.

I wanted to share what I thought, but honestly it's not a final opinion, as I know the advantages of leaving it unified. With these points what do you think? If you prefer, I can adapt the code to use `historical_predictions.pkl`.

Reply:
I don’t quite understand. Why do you say it “brings all candles?” It’s all sitting in the `self.dd.historic_prédictions` Dictionary, which is in ram. It has the datetime objects, so you can merge or slice or take the last candle, whatever you need. Is there some performance issue with using self.dd.historic_predictions?

Why do you say “it is per pair?” Self.dd.historic_predictions is also a dictionary where each key is “per pair” to access the per pair dataframes. Can you explain why this is not a sufficient structure so as to warrant writing new files per pair?

Reply:
Code updated to use `self.dd.historic_prédictions`